### PR TITLE
Cosmins long running scenario with some config changes to adapt

### PR DESCRIPTION
**Pull request by agatsoh**
Fixes: #5177 

## Description

Please, describe what this PR does in detail:
- Scenario introduces back the PR https://github.com/raiden-network/raiden/pull/2763
- The PR tries to mimic the users action of keeping the raiden node alive for a sufficiently long amount of time to check whether some bugs are introduced with a sufficiently dirty blockchain state.
- The PR does not yet introduce fees(of anykind) or interact with MS. Though it interacts with the PFS for routes.
- The scenario could be run independently by users/testers/developers to test out where it fails.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+          tasks:
+            - close_channel: {from: 4, to: 2}
+            - wait: 20
+            - assert: {from: 2, to: 4, total_deposit: 150, balance: 100, state: "closed"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 50, state: "closed"}
+              # - serial:
+        # name: "199 payments from A to D in a channel"
+        # repeat: 199
+        # tasks:
+      - transfer: {from: 0, to: 3, amount: 200}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Here you're sending 200 to node3 through node2.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+      - wait: 5
+      - serial:
+          name: "Let the node get into a stable state"
+          tasks:
+            - wait: 60
+      - parallel:
+          name: "Assert channels after transfers"
+          tasks:
+            - assert: {from: 2, to: 3, total_deposit: 101, balance: 1, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101, balance: 201, state: "opened"}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: But here you're only asserting that 199 was sent, and not 200.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+    environment-type: development
+    default-settle-timeout: 40
+    default-reveal-timeout: 20
+    proportional-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+    proportional-imbalance-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Could add a description of what the scenario does like all other scenarios have? And also add it to the README?

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+      - parallel:
+          name: "Create channels"
+          tasks:
+            - open_channel: {from: 0, to: 1, total_deposit: 0, expected_http_status: 201}
+            - open_channel: {from: 1, to: 2, total_deposit: 0, expected_http_status: 201}
+            - open_channel: {from: 2, to: 3, total_deposit: 0, expected_http_status: 201}
+      - serial:
+          name: "Allow time for nodes to learn about their channels"
+          tasks:
+            - wait: 30
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This should not be necessary, but there's no harm in leaving it in other than that it takes a bit of time.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+      - parallel:
+          name: "Verify if channels were opened correctly"
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 0, balance: 0,  state: "opened"}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Could you in general use the same amounts as all other scenarios? All other scenarios are using values of around 10**18 tokens for deposits.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+            - assert: {from: 0, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 3, total_deposit: 101, balance: 101, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101, balance: 101, state: "opened"}
+      - parallel:
+          name: "Deposit to a channel that does not exist"
+          tasks:
+            # Node tries to deposit to an unexisting channel
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: unexisting -> non existing

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+            - deposit: {from: 4, to: 3, total_deposit: 20, expected_http_status: 409}
+            # Node tries to deposit less then the total deposit in two of his channels
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: less then -> less than

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+            - deposit: {from: 1, to: 0, total_deposit: 50, expected_http_status: 409}
+            - deposit: {from: 1, to: 2, total_deposit: 50, expected_http_status: 409}
+            # Opening a channel, but the channel already exists
+            - open_channel: {from: 1, to: 0, total_deposit: 100, expected_http_status: 409}
+      - serial:
+          name: "Give the nodes time to sync"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This formulation is a bit weird. What is meant by syncing?

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+          name: "Checking for the channel state to be unchanged"
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 3, total_deposit: 101, balance: 101, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101, balance: 101, state: "opened"}
+      - serial:
+          name: "C tries different cases"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: We have agreed to use numbers instead of letters to describe the nodes in general, since that's what the scenario player uses internally and this makes it easier to reason about.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+          tasks:
+            - open_channel: {from: 2, to: 4, total_deposit: 0, expected_http_status: 201}
+            - wait: 10
+            - deposit: {from: 2, to: 4, total_deposit: 100}
+            - wait: 10
+            # Try to make a deposit smaller the the amount that was deposited
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: the the -> than the

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 11]
```diff
@@ -0,0 +1,335 @@
...
+            - deposit: {from: 2, to: 4, total_deposit: 30, expected_http_status: 409}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: You're already trying to do this in line 82 and 83. I don't think it's necessary to do it again.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 12]
```diff
@@ -0,0 +1,335 @@
...
+      - serial:
+          name: "Restarting node B"
+          tasks:
+            - start_node: 1
+      - parallel:
+          name: "Checking for the D <-> C channel state"
+          tasks:
+            - assert: {from: 2, to: 3, total_deposit: 101, balance: 101, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101, balance: 101, state: "opened"}
+              #      - serial:
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: If you don't want this part in the scenario, then please remove it instead of leaving it as a comment.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 0, state: "opened"}
+      - serial:
+          name: "Closing down channel E <-> C"
+          tasks:
+            - close_channel: {from: 4, to: 2}
+            - wait: 20
+            - assert: {from: 2, to: 4, total_deposit: 150, balance: 100, state: "closed"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 50, state: "closed"}
+              # - serial:
+        # name: "199 payments from A to D in a channel"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Please remove this part if it's not going to be used

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+            - wait: 5
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 11, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 189, state: "opened"}
+            # B sends 10 tokens to A
+            - transfer: {from: 1, to: 0, amount: 10, lock_timeout: 30}
+            - wait: 5
+            - assert: {from: 0, to: 1, total_deposit: 260, balance: 10, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100, balance: 299, state: "opened"}
+            # E sends 10 tokens to C (A -> B-> C)
+            - transfer: {from: 4, to: 2, amount: 10, lock_timeout: 30}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This would have an expected http code of 409, right?

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 5 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: Done

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 6 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: Changed

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 13 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: Removed

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 3 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: Will add it.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 15]
```diff
@@ -0,0 +1,327 @@
...
+    pfs:
+      url: https://pfs-goerli.services-dev.raiden.network
+    udc:
+      enable: true
+      token:
+        deposit: true
+
+token:
+  address: "0x62083c80353Df771426D209eF578619EE68D5C7A"
+  balance_fund: 200_000_000_000_000_000_000
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: 200 ETH is probably quite a lot more than you need. You could just have copy pasted the values from the other scenarios. I don't think it's very realistic to have people depositing 100 ETH into a payment channel.

File: raiden/tests/scenarios/README.md [hunk 16]
```diff
@@ -27,6 +27,13 @@ It is also tested that a node can be stopped and started again and that it still
 Several nodes perform up to 100 payments.
 In the end it is tested that channels can be closed and that the monitoring service correctly kicks in if a node is offline during closing.
 
+#### [bf2_long_running](./bf2_long_running.yaml)
+
+This is the bf2 long running scenario. This scenario mimics user behaviour for
```
Comment (czepluch) [inline raiden/tests/scenarios/README.md]: Please, delete the first sentence here in order to be consistent with the other descriptions.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 17]
```diff
@@ -0,0 +1,332 @@
...
+    proportional-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+    proportional-imbalance-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+
+# This is the bf2 long running scenario. This scenario mimics user behaviour for 
+# opening channels, depositing, transferring, waiting keeping the raiden node alive for 
+# a long time in the process checking whether all distributed system in raiden network
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: What is meant by "all distributed system"?

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 18]
```diff
@@ -0,0 +1,332 @@
...
+          name: "Verify if channels were opened correctly"
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 0, balance: 0,  state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 0, balance: 0,  state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 0, balance: 0,  state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 0, balance: 0,  state: "opened"}
+            - assert: {from: 2, to: 3, total_deposit: 0, balance: 0,  state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 0, balance: 0,  state: "opened"}
+      - parallel:
+          name: "0 1 2 3 make a deposit of 100 to their channels"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: `100` here doesn't make any sense anymore.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 19]
```diff
@@ -0,0 +1,332 @@
...
+          tasks:
+            - deposit: {from: 0, to: 1, total_deposit: 100_000_000_000_000_000}
+            - deposit: {from: 1, to: 0, total_deposit: 100_000_000_000_000_000}
+            - deposit: {from: 1, to: 2, total_deposit: 100_000_000_000_000_000}
+            - deposit: {from: 2, to: 1, total_deposit: 100_000_000_000_000_000}
+            - deposit: {from: 2, to: 3, total_deposit: 101_000_000_000_000_000}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This amount is different from the others. Maybe add a comment noting it and telling why.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 20]
```diff
@@ -0,0 +1,332 @@
...
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 2, to: 3, total_deposit: 101_000_000_000_000_000, balance: 101_000_000_000_000_000, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101_000_000_000_000_000, balance: 101_000_000_000_000_000, state: "opened"}
+      - parallel:
+          name: "Deposit to a channel that does not exist"
+          tasks:
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Just a note: These are also tested in the api integration tests, but I think it's fine to keep them in for now, for the sake of having a scenario where a lot of stuff is going on.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 21]
```diff
@@ -0,0 +1,332 @@
...
+            # Node tries to deposit to an non-existing channel
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: `an` -> `a`

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 22]
```diff
@@ -0,0 +1,332 @@
...
+            - deposit: {from: 4, to: 3, total_deposit: 20_000_000_000_000_000, expected_http_status: 409}
+            # Node tries to deposit less than the total deposit in two of his channels
+            - deposit: {from: 1, to: 0, total_deposit: 50_000_000_000_000_000, expected_http_status: 409}
+            - deposit: {from: 1, to: 2, total_deposit: 50_000_000_000_000_000, expected_http_status: 409}
+            # Opening a channel, but the channel already exists
+            - open_channel: {from: 1, to: 0, total_deposit: 100_000_000_000_000_000, expected_http_status: 409}
+      - serial:
+          name: "Give the nodes time to complete the earlier deposits and channel opening"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This really shouldn't be necessary. I'd probably delete this entire section of waiting for earlier deposits since they tasks above should wait for the API calls to finish anyway.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 23]
```diff
@@ -0,0 +1,332 @@
...
+          name: "Checking for the channel state to be unchanged"
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "opened"}
+            - assert: {from: 2, to: 3, total_deposit: 101_000_000_000_000_000, balance: 101_000_000_000_000_000, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101_000_000_000_000_000, balance: 101_000_000_000_000_000, state: "opened"}
+      - serial:
+          name: "2 tries different cases"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Can you be more specific about what cases are tried? This would give the reader a quick overview of what to expect to happen.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 24]
```diff
@@ -0,0 +1,332 @@
...
+            # New channel
+            - assert: {from: 2, to: 4, total_deposit: 150_000_000_000_000_000, balance: 150_000_000_000_000_000, state: "opened"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 0, state: "opened"}
+      - serial:
+          name: "Continue building the network"
+          tasks:
+            # 0 opens a channel with 4
+            - open_channel: {from: 0, to: 4, total_deposit: 10_000_000_000_000_000, expected_http_status: 201}
+            - assert: {from: 0, to: 4, total_deposit: 10_000_000_000_000_000, balance: 10_000_000_000_000_000, state: "opened"}
+            # 4 deposits 25 tokens onto the 0 - 4 channel
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: It's more than 25, right?

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 25]
```diff
@@ -0,0 +1,332 @@
...
+            - deposit: {from: 4, to: 0, total_deposit: 25_000_000_000_000_000}
+            - wait: 20
+            # 0 performs a payment to 4 of 50 tokens (path 0<->1<->2<->4)
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Again here. Please update the value. 

And can you please in general write `node0` or `nodeX` instead of just 0 or 1? This makes it easier to read and it complies with how it's done in all other scenarios.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 26]
```diff
@@ -0,0 +1,332 @@
...
+          tasks:
+            - start_node: 1
+      - parallel:
+          name: "Checking for the 3 <-> 2 channel state"
+          tasks:
+            - assert: {from: 2, to: 3, total_deposit: 101_000_000_000_000_000, balance: 101_000_000_000_000_000, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101_000_000_000_000_000, balance: 101_000_000_000_000_000, state: "opened"}
+            - assert: {from: 0, to: 1, total_deposit: 100_000_000_000_000_000, balance: 40_000_000_000_000_000, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100_000_000_000_000_000, balance: 160_000_000_000_000_000, state: "opened"}
+      - transfer: {from: 3, to: 2, amount: 99_000_000_000_000_000, lock_timeout: 30}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This transfer seems to be a bit dislocated. Can you please add it in a `tasks` section and explain what it does?

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 27]
```diff
@@ -0,0 +1,332 @@
...
+            - assert: {from: 1, to: 2, total_deposit: 100_000_000_000_000_000, balance: 200_000_000_000_000_000, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100_000_000_000_000_000, balance: 0, state: "opened"}
+      - serial:
+          name: "Closing down channel 4 <-> 2"
+          tasks:
+            - close_channel: {from: 4, to: 2}
+            - wait: 20
+            - assert: {from: 2, to: 4, total_deposit: 150_000_000_000_000_000, balance: 100_000_000_000_000_000, state: "closed"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 50_000_000_000_000_000, state: "closed"}
+      - transfer: {from: 0, to: 3, amount: 200_000_000_000_000_000, lock_timeout: 30}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This is also a bit misplaced it seems

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 28]
```diff
@@ -0,0 +1,332 @@
...
+      - wait: 5
+      - serial:
+          name: "Let the node get into a stable state"
+          tasks:
+            - wait: 60
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Why wouldn't the node be stable? Why do you think this is needed?

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 29]
```diff
@@ -0,0 +1,332 @@
...
+          tasks:
+            - close_channel: {from: 4, to: 0}
+            - wait: 5
+      - parallel:
+          name: "Check if channel was closed"
+          tasks:
+            - assert: {from: 4, to: 0, total_deposit: 25_000_000_000_000_000, balance: 15_000_000_000_000_000, state: "closed"}
+            - assert: {from: 0, to: 4, total_deposit: 10_000_000_000_000_000, balance: 20_000_000_000_000_000, state: "closed"}
+      - parallel:
+          name: "All nodes try to make a payment to 4"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Can you add why it's not expected to work?

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 30]
```diff
@@ -0,0 +1,332 @@
...
+            - transfer: {from: 2, to: 4, amount: 1_000_000_000_000_000, expected_http_status: 409, lock_timeout: 30}
+            - transfer: {from: 3, to: 4, amount: 1_000_000_000_000_000, expected_http_status: 409, lock_timeout: 30}
+      - serial:
+          name: "0 and 1 leave token netwwork"
+          tasks:
+            - leave_network: {from: 0}
+            - leave_network: {from: 1}
+            - wait: 20
+      - serial:
+          name: "Check if the only channel open is 2 <-> 3"
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: You don't check if it's the only channel open. You just check that it is open. Not that none of the others are not.

**Comment by agatsoh** [comment]
According to our discussion yesterday, Just to put it on record that
 I again ran the scenario and I again encounter the same No route available error at this place
```
- transfer: {from: 0, to: 4, amount: 50_000_000_000_000_000, expected_http_status: 200, lock_timeout: 30}
```
line 139
What I observe through all these runs is that the failures for 
`Payment couldn't be completed because: there is no route available` exception is quite random and can occur at any time during the scenario even right the end.

**Comment by LefterisJP** [comment]
@agatsoh was that after https://github.com/raiden-network/raiden/pull/5339 was merged?

**Comment by czepluch** [comment]
That PR by Karl, doesn't touch this scenario.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 31]
```diff
@@ -0,0 +1,335 @@
...
+            # node2 tries to send another tokens payment to node1 (fail no route with enough capacity)
+            - transfer: {from: 2, to: 1, amount: 150_000_000_000_000_000, expected_http_status: 409, lock_timeout: 30}
+            - wait: 100
+      - serial:
+          name: "Multiple multi hop payments"
+          repeat: 10
+          tasks:
+            # node2 sends 10 payments to node1 by using the 2 <-> 4 <-> 0 <-> 1 route
+            - transfer: {from: 2, to: 1, amount: 1_000_000_000_000_000, lock_timeout: 30}
+            - wait: 100
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Please remove this `wait`. It is run for every transfer! And you have a dedicated wait right after the 10 transfers, which is enough.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 32]
```diff
@@ -0,0 +1,335 @@
...
+          name: "Starting closing edge cases"
+          tasks:
+            # node2 tries to make a deposit to the channel that is being closed (fail 409)
+            - deposit: {from: 2, to: 4, total_deposit: 200_000_000_000_000_000, expected_http_status: 409}
+            - wait: 100
+            # node2 sends tokens to node1
+            - transfer: {from: 2, to: 1, amount: 10_000_000_000_000_000, lock_timeout: 30}
+            - wait: 100
+            - assert: {from: 1, to: 2, total_deposit: 100_000_000_000_000_000, balance: 10_000_000_000_000_000, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100_000_000_000_000_000, balance: 189_000_000_000_000_000, state: "opened"}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: According to the deposit above and the transfer of 200_000_000_000_000_000, this assert is wrong. It should be `190` not `189`.

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 33]
```diff
@@ -0,0 +1,335 @@
...
+            # node1 sends tokens to node0
+            - transfer: {from: 1, to: 0, amount: 10_000_000_000_000_000, lock_timeout: 30}
+            - wait: 100
+            - assert: {from: 0, to: 1, total_deposit: 260_000_000_000_000_000, balance: 10_000_000_000_000_000, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100_000_000_000_000_000, balance: 299_000_000_000_000_000, state: "opened"}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: This is also wrong. Should be `300`, not `299`

File: raiden/tests/scenarios/bf2_long_running.yaml [hunk 34]
```diff
@@ -0,0 +1,335 @@
...
+            # node4 sends tokens to node2 (0 -> 1-> 2)
+            - transfer: {from: 4, to: 2, amount: 10_000_000_000_000_000, lock_timeout: 30}
+            - wait: 100
+      - parallel:
+          name: "Assert for the channel states after mediated transfers were completed"
+          tasks:
+            - assert: {from: 1, to: 2, total_deposit: 100_000_000_000_000_000, balance: 0, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100_000_000_000_000_000, balance: 200_000_000_000_000_000, state: "opened"}
+            - assert: {from: 0, to: 1, total_deposit: 260_000_000_000_000_000, balance: 0, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100_000_000_000_000_000, balance: 309_000_000_000_000_000, state: "opened"}
```
Comment (czepluch) [inline raiden/tests/scenarios/bf2_long_running.yaml]: Can you please go through all these asserts and make sure the calculations are correct?

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 33 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: It is actually 350

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 32 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: done

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 31 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: removed

File: raiden/tests/scenarios/bf2_long_running.yaml (hunk 34 above)
Comment (agatsoh) [reply raiden/tests/scenarios/bf2_long_running.yaml]: Done

**Comment by czepluch** [comment]
Did the scenario pass locally?

**Comment by agatsoh** [comment]
Hello @christianbrb and @czepluch  the Scenario ran successfully for me so that now we could merge the PR if it ok.

**Comment by palango** [comment]
@agatsoh Nice PR, for the future. There's a lot of small commits: can you squash them to make the git history more useful?

**Comment by agatsoh** [comment]
@palango Yes will definitely squash the commits
